Mint cache

ipmb opened the pull request:
Inspired by https://www.djangosnippets.org/snippets/155/ and borrowed
from https://github.com/ericflo/django-newcache

* [x] test coverage
* [x] docs

ipmb committed:
Mint cache

Inspired by https://www.djangosnippets.org/snippets/155/ and borrowed
from https://github.com/ericflo/django-newcache

ipmb committed:
Add missing test dependency

unbracketed reviewed (approved):
Neat!  The code looks good from a read through.

A code comment referencing `MintCache` and `newcache` would help others know where the naming and idea came from

ipmb committed:
💯  coverage

ipmb committed:
Codecov

ipmb committed:
Modify how coverage is run

ipmb committed:
Test with both mixins

ipmb committed:
Support back to Django 1.5

ipmb committed:
Don't test legacy versions on py35

ipmb committed:
Revert "Don't test legacy versions on py35"

This reverts commit a0a06ce97598fba1a93086152613975e67151fc1.

ipmb committed:
Changes for travis

ipmb committed:
Django 2.0 now returns a value from set_many

https://github.com/django/django/commit/a027447f5606d60358e5da76c0e8233732ad2b17

ipmb committed:
Docs for mint cache
